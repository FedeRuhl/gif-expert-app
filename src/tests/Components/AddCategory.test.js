import '@testing-library/jest-dom';
import { shallow } from 'enzyme'
import AddCategory from '../../Components/AddCategory';
import React from 'react';

describe('Add Category Component test', () => {

    const setCategories = jest.fn();
    let wrapper;

    beforeEach(() => {
        jest.clearAllMocks();
        wrapper = shallow(<AddCategory setCategories={ setCategories }/>);
    });
    
    test('should return the correct form', () => {
        expect(wrapper).toMatchSnapshot();
    });

    test('should change textbox input', () => {
        let input = wrapper.find('input');
        const value = 'Hello world';

        input.simulate('change', {
            target: {
                value: value
            }
        });

        const inputRefreshed = wrapper.find('input');
        expect(inputRefreshed.prop('value')).toBe(value);
    });
    
    test('should not call setCategories method if inputValue length is minor or equal than 2', () => {
        
        wrapper.find('form')
            .simulate('submit', { preventDefault(){} });

        expect(setCategories).not.toHaveBeenCalled();
    });

    test('should call setCategories method and clear the textbox input if inputValue length is greater than 2', () => {

        let input = wrapper.find('input');
        const value = 'Hello world';

        input.simulate('change', {
            target: {
                value: value
            }
        });

        wrapper.find('form')
            .simulate('submit', { preventDefault(){} });
        
        expect(setCategories).toHaveBeenCalled();
        expect(setCategories).toHaveBeenCalledWith(expect.any(Function));
            
        input = wrapper.find('input');
        expect(input.prop('value')).toBe('');
    });
        

});