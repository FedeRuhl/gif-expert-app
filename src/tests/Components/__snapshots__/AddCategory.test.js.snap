// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Add Category Component test should return the correct form 1`] = `
<form
  onSubmit={[Function]}
>
  <input
    onChange={[Function]}
    type="text"
    value=""
  />
</form>
`;
